### DISCUSSION TITLE
How to get camera param (especially for depth) for 3D pointcloud alignment in latest scripthook?

### DISCUSSION THREAD

**Discussion by choiszt** [scripthookvdotnet/scripthookvdotnet#1335] [Q&A]
Hi @kagikn !
I'm currently attempting to align point cloud data from RGBD collected in GTA-V. However, I'm struggling to find the appropriate intrinsic matrices. Moreover, I've been unable to locate the extrinsic matrix necessary to accurately transform the depth image to world coordinates. I suspect the depth images might not be in the camera coordinates, which means I can't directly apply the camera parameters for the transformation. Are there any chances in your latest powerful scripthook to acquire the correct intrinsic matrix of default camera and the transformation matrix of from depth coords to World?(I guess I miss the NDC to World matrix) Any insights or experience you could share would be greatly appreciated. Thank you in advance!

**Comment by kagikn** [comment]
Hi!
I appreciate your interest in SHVDN. I'm afraid that I don't have sufficient knowledge to find intrinsic or extrinsic matrices for cameras. I don't have enough information about how exactly camera classes works (in the exe), and I'm only sure a bit of what they store, such as how they store their transformation matrices (try `Camera.MatrixAddress` for such matrices). I'm not good at algebra, either, and I have no idea how to find intrinsic or extrinsic matrices (I didn't hear of such terms until now even).

Maybe someone might be able to answer, but unfortunately I don't have any answers.
